libsolidity/ast/Types.cpp

```diff
@@ -412,7 +412,9 @@ BoolResult AddressType::isImplicitlyConvertibleTo(Type const& _other) const
 
 BoolResult AddressType::isExplicitlyConvertibleTo(Type const& _convertTo) const
 {
-	if (auto const* contractType = dynamic_cast<ContractType const*>(&_convertTo))
+	if (_convertTo.category() == category())
```

Comment:
Isn't this caught by `isImplicitlyConvertibleTo`?

Reply:
Ah sorry no!

Reply:
``address payable`` is *not* implicitly convertible to ``address``